### PR TITLE
[15.0][IMP] report_qweb_pdf_watermark: Improvements

### PR DESCRIPTION
**Pull request by bosd**

### CODE REVIEW

**Comment by bosd** [comment]
@espo-tony Can you review?

File: report_qweb_pdf_watermark/models/report.py [hunk 1]
```diff
@@ -115,11 +115,11 @@ def _run_wkhtmltopdf(
...
                 logger.exception("Failed to load watermark", e)
 
-        if not pdf_watermark:
-            logger.error("No usable watermark found, got %s...", watermark[:100])
+        if not self.pdf_has_usable_pages(pdf_watermark):
             return result
 
-        if not self.pdf_has_usable_pages(pdf_watermark.numPages):
+        if not pdf_watermark:
+            logger.error("No usable watermark found, got %s...", watermark[:100])
```
Comment (thomaspaulb) [inline report_qweb_pdf_watermark/models/report.py:122]: You've swapped the two checks around, but shouldn't the check for `if not pdf_watermark` come first? The other check actually goes into pdf_watermark.numPages, so seems like it should come after.

File: report_qweb_pdf_watermark/models/report.py [hunk 2]
```diff
@@ -48,11 +48,11 @@ def _render_qweb_pdf(self, res_ids=None, data=None):
...
         return super(Report, self)._render_qweb_pdf(res_ids=res_ids, data=data)
 
-    def pdf_has_usable_pages(self, numpages):
-        if numpages < 1:
+    def pdf_has_usable_pages(self, pdf_watermark):
+        if pdf_watermark.numPages < 1:
             logger.error("Your watermark pdf does not contain any pages")
             return False
-        if numpages > 1:
+        if pdf_watermark.numPages > 1:
```
Comment (thomaspaulb) [inline report_qweb_pdf_watermark/models/report.py:55]: This refactor doesn't really do anything

**Review by thomaspaulb — CHANGES_REQUESTED** [review]

**Comment by thomaspaulb** [comment]
Is this a cherry-pick of a fix for another version?

**Comment by bosd** [comment]
> Is this a cherry-pick of a fix for another version?

Yes, These are improvements made by @espo-tony.

**Comment by thomaspaulb** [comment]
Can you help out with a link to the original PR

**Comment by bosd** [comment]
> Can you help out with a link to the original PR

No, as these changes were made in private repo. 
For that reason I'm committing it here.  :smile: 

>This refactor doesn't really do anything
True, maybe better code practice/readability of the code.

Feel free to close/merge close this pr.

File: report_qweb_pdf_watermark/tests/test_report_qweb_pdf_watermark.py [hunk 3]
```diff
@@ -42,14 +42,15 @@ def _test_report_images(self, number):
         )
         self.assertEqual(pdf.count(b"/Subtype /Image"), number)
 
-    def test_pdf_has_usable_pages(self):
-        # test 0
-        numpages = 0
+    def _test_pdf_has_usable_pages(self, pdf_watermark):
```
Comment (thomaspaulb) [inline report_qweb_pdf_watermark/tests/test_report_qweb_pdf_watermark.py:45]: Also, this test is disabled?!

**Comment by thomaspaulb** [comment]
@bosd Yes, it doesn't seem to me that this does anything useful. So I'll close it.

**Pull request closed without merging**